pinot-tools/src/main/java/org/apache/pinot/tools/realtime/provisioning/MemoryEstimator.java

```diff
@@ -387,4 +418,182 @@ private long calculateMemoryForCompletedSegmentsPerPartition(long completedSegme
   public String[][] getNumSegmentsQueriedPerHost() {
     return _numSegmentsQueriedPerHost;
   }
+
+  private static File generateCompletedSegment(File dataCharacteristicsFile, File schemaFile, TableConfig tableConfig) {
```

Comment:
Please make sure that dataCharacteristicsFile format is the exact same json file format that is taken by the recommendation engine (RecommenderDrivr.java). Otherwise, the follow up integration task won't work.

Reply:
separate schemaFile should not be needed